core/src/main/java/org/apache/brooklyn/enricher/stock/AbstractTransformer.java

```diff
@@ -57,14 +57,16 @@ public AbstractTransformer() {
     public void setEntity(EntityLocal entity) {
         super.setEntity(entity);
 
-        Function<SensorEvent<T>, U> transformation = getTransformation();
         this.producer = getConfig(PRODUCER) == null ? entity: getConfig(PRODUCER);
         this.sourceSensor = (Sensor<T>) getRequiredConfig(SOURCE_SENSOR);
         Sensor<?> targetSensorSpecified = getConfig(TARGET_SENSOR);
         this.targetSensor = targetSensorSpecified!=null ? (Sensor<U>) targetSensorSpecified : (Sensor<U>) this.sourceSensor;
         if (producer.equals(entity) && targetSensorSpecified==null) {
+            // We cannot call getTransformation() here to log the tranformation, as it will attempt
+            // to resolve the transformation, which will cause the entity initialization thread to
+            // block
             LOG.error("Refusing to add an enricher which reads and publishes on the same sensor: "+
```

Comment:
As stated in the comment, we can't call getTransformation() here as the entity is being initialized, so the call will block